paparazzi/src/main/java/app/cash/paparazzi/PaparazziTestExtension.kt

```diff
@@ -0,0 +1,465 @@
+/*
+ * Copyright (C) 2019 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.paparazzi
+
+import android.content.Context
+import android.content.res.Resources
+import android.graphics.Bitmap
+import android.util.AttributeSet
+import android.util.DisplayMetrics
+import android.view.*
+import androidx.annotation.LayoutRes
+import app.cash.paparazzi.agent.InterceptorRegistrar
+import app.cash.paparazzi.internal.*
+import app.cash.paparazzi.internal.parsers.LayoutPullParser
+import com.android.ide.common.rendering.api.RenderSession
+import com.android.ide.common.rendering.api.Result
+import com.android.ide.common.rendering.api.Result.Status.ERROR_UNKNOWN
+import com.android.ide.common.rendering.api.SessionParams
+import com.android.layoutlib.bridge.Bridge.cleanupThread
+import com.android.layoutlib.bridge.Bridge.prepareThread
+import com.android.layoutlib.bridge.BridgeRenderSession
+import com.android.layoutlib.bridge.impl.RenderAction
+import com.android.layoutlib.bridge.impl.RenderSessionImpl
+import com.android.tools.layoutlib.java.System_Delegate
+import org.junit.jupiter.api.extension.AfterEachCallback
+import org.junit.jupiter.api.extension.BeforeEachCallback
+import org.junit.jupiter.api.extension.ExtensionContext
+import java.awt.image.BufferedImage
+import java.lang.reflect.Field
+import java.lang.reflect.Modifier
+import java.util.*
+import java.util.concurrent.TimeUnit
+import kotlin.collections.LinkedHashSet
+import kotlin.collections.find
+import kotlin.collections.forEach
+import kotlin.collections.setOf
+
+class PaparazziTestExtension(
```

Comment:
This class is a copy of `Paparazzi` with the JUnit 4 `TestRule` implementation replaced with JUnit 5 `Extension` implementations.

Reply:
I now realize I can probably put both the JUnit 4 & 5 code in the main `Paparazzi` class if I move some things around. That would be a lot less duplication and both versions could use the same class.

Reply:
Removed.